### PR TITLE
consume APIs to search tool shed repositories and tools

### PR DESCRIPTION
**Pull request by martenson**

### CODE REVIEW

**Comment by nsoranzo** [comment]
You have some trailing whitespaces that break TravisCI, can you fix them? I'm fine with a squashed commit if you like.

**Comment by martenson** [comment]
@nsoranzo do you think it is ok now?

File: bioblend/toolshed/repositories/__init__.py [hunk 1]
```diff
@@ -35,6 +35,72 @@ def get_repositories(self):
...
         return Client._get(self)
 
+    def search_repositories(self, q, page=1, page_size=10):
+        """
+        Search for repositories in a Galaxy Tool Shed
+
+        :type  q: str
+        :param q: query string for searching purposes
+
+        :type  page: str
```
Comment (nsoranzo) [inline bioblend/toolshed/repositories/__init__.py]: Should this be an `int`?

File: bioblend/toolshed/repositories/__init__.py [hunk 2]
```diff
@@ -35,6 +35,72 @@ def get_repositories(self):
...
+        :param page: page requested
+
+        :type  page_size: str
```
Comment (nsoranzo) [inline bioblend/toolshed/repositories/__init__.py]: As above, should this be an `int`?

File: bioblend/toolshed/tools/__init__.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        super(ToolShedClient, self).__init__(toolshed_instance)
+
+    def search_tools(self, q, page=1, page_size=10):
+        """
+        Search for tools in a Galaxy Tool Shed
+
+        :type  q: str
+        :param q: query string for searching purposes
+
+        :type  page: str
```
Comment (nsoranzo) [inline bioblend/toolshed/tools/__init__.py]: Should this be an `int`?

File: bioblend/toolshed/tools/__init__.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        :param page: page requested
+
+        :type  page_size: str
```
Comment (nsoranzo) [inline bioblend/toolshed/tools/__init__.py]: Should this be an `int`?

File: bioblend/toolshed/repositories/__init__.py (hunk 1 above)
Comment (martenson) [inline bioblend/toolshed/repositories/__init__.py]: It is being cast to float anyways so 1 and '1'  should both work. Do you think putting int here makes more sense? If so I will change it.

Comment (nsoranzo) [inline bioblend/toolshed/repositories/__init__.py]: Yes, please, we do the same for boolean parameters.

**Comment by martenson** [comment]
I forgot about this, sorry, fixed now
ping @nsoranzo

**Comment by nsoranzo** [comment]
@martenson Thanks for the fixes, but I think that the merge of master is not correct, e.g. the changes to `bioblend/toolshed/__init__.py` are lost. Maybe rebase on master?

**Comment by martenson** [comment]
@nsoranzo great catch! Sorry about that :(

**Comment by nsoranzo** [comment]
Thanks @martenson!

**Pull request merged**